### PR TITLE
Added boosting for certain series

### PR DESCRIPTION
**Pull request by mhilton**
Added boosts to trusty > precise > utopic > everything else.

Note I implemented this as described on the card so there is specifically no boosting to bundles, That might want to be looked at. I have made the series boost lower than the promulgated boost so all things being equal for any given charm you should see the promulgated trusty charm at the top of the list followed by the promulgated precise one etc.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/539/

**Comment by mhilton** [comment]
## QA Instructions
1. Make sure elasticsearch has an up to date configuration.
   
   cd ${GOPATH}/src/github.com/juju/charmstore
   make install
   http DELETE :9200/charmstore
   essync -settings=internal/elasticsearch/definitions/index.json -mapping=internal/elasticsearch/definitions/entity.json &lt;path/to/config.yaml&gt;
2. Perform search.
   
   charmd &lt;path/to/config.yaml&gt; &amp;
   http :8080/v4/search?text=mysql

Expected results should be promulgated charms first, followed by user charms. Within those sets trusty charms should come first followed by precise, then utopic, then everything else.

Note the result ordering may not be exact as it depends on other factors to do with the content of the charms.

File: internal/charmstore/search.go [hunk 1]
```diff
@@ -33,6 +34,13 @@ type esDoc struct {
 	Series string
 }
 
+// seriesBoost is a map defining how much the results for a particular series should be boosted
```
Comment (rogpeppe) [inline internal/charmstore/search.go]: // seriesBoost defines how much the results for each
// series will be boosted.

?

File: internal/charmstore/search_test.go [hunk 2]
```diff
@@ -431,6 +431,20 @@ func (s *StoreSearchSuite) TestSorting(c *gc.C) {
...
 
+func (s *StoreSearchSuite) TestBoosting(c *gc.C) {
+	err := s.store.ExportToElasticSearch()
+	c.Assert(err, gc.IsNil)
+	s.store.ES.Database.RefreshIndex(s.TestIndex)
+	var sp SearchParams
+	res, err := s.store.Search(sp)
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.Results, gc.HasLen, 4)
+	c.Assert(res.Results[0].String(), gc.Equals, exportTestCharms["mysql"])
```
Comment (rogpeppe) [inline internal/charmstore/search_test.go]: c.Assert(res.Results, gc.DeepEquals, []*charm.Reference{
        exportTestCharms["mysql"],
        exportTestCharms["wordpress"],
        exportTestCharms["wordpress-simple",
        exportTestCharms["varnish"],
   }

?

File: internal/charmstore/search.go [hunk 3]
```diff
@@ -33,6 +34,13 @@ type esDoc struct {
...
+var seriesBoost = map[string]float64{
+	"trusty":  1.125,
+	"precise": 1.1125,
+	"utopic":  1.1,
```
Comment (bac) [inline internal/charmstore/search.go:44]: Perhaps a note about the ranking being used. I assume it is LTS in reverse order followed by non-LTS releases?

**Comment by bac** [comment]
The code is :+1: with Roger's suggestions.  QA is :+1:  Thanks for the clear instructions.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/540/

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**